### PR TITLE
Airtable prop defs

### PR DESCRIPTION
**Pull request by malexanderlim**
### <samp>🤖 Generated by Copilot at e893435</samp>

Improved the props and descriptions of the airtable app and sources to enhance the user experience and reduce code duplication. Added two new app components, `assemblyai` and `senta`, to enable speech recognition and sentiment analysis integrations. Updated the `pnpm-lock.yaml` file accordingly.

### <samp>🤖 Generated by Copilot at e893435</samp>

> _We are the masters of the airtable app_
> _We unleash the power of the props_
> _We integrate with APIs of speech and sentiment_
> _We are the pipedream of the code_

### <samp>🤖 Generated by Copilot at e893435</samp>

*  Moved limiter and axiosRateLimiter variables from source to app component, to enable rate limiting for all airtable sources and actions ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-62cbf0e8592386d74ff0a9eec1b3c5c8cf12c1270824be12242984580c5f8ab2L8-R8), )
* Added optional prop to view prop in app component, to allow creating and updating records without specifying a view ID ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-aab74c687a9d097b45af4c3706724cc2cab9723fd7b8af0b2176d2b2ae912ebbR135))
* Updated descriptions of table and view props in app component, to use more specific examples of dynamic references from trigger event metadata ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-aab74c687a9d097b45af4c3706724cc2cab9723fd7b8af0b2176d2b2ae912ebbL25-R25), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-aab74c687a9d097b45af4c3706724cc2cab9723fd7b8af0b2176d2b2ae912ebbL43-R43))
* Incremented versions of all airtable sources, to reflect changes in props and code ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-62cbf0e8592386d74ff0a9eec1b3c5c8cf12c1270824be12242984580c5f8ab2L16-R16), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-b94c03b1bcdcfc3f77caa216a2a56f5d623c7e54cc994fce2c6d8af1d3806efaL11-R11), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-2ae730a6527e2bd1c97044a4f0bbe73b52df72a478f265914426c09499609325L11-R11), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-349481156959f6eed80b03e1c58b5fd94e08c83eec176b4c95f5eb560a607234L11-R11), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-7a48f7edc8054aad6e6eac7518f6b4bb3b5a33af9687eb69a11ef91dc54ac077L11-R11))
* Added descriptions of table and view props in all airtable sources, to provide more information and consistency ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-62cbf0e8592386d74ff0a9eec1b3c5c8cf12c1270824be12242984580c5f8ab2R30), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-b94c03b1bcdcfc3f77caa216a2a56f5d623c7e54cc994fce2c6d8af1d3806efaR23), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-b94c03b1bcdcfc3f77caa216a2a56f5d623c7e54cc994fce2c6d8af1d3806efaR36), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-2ae730a6527e2bd1c97044a4f0bbe73b52df72a478f265914426c09499609325R23), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-349481156959f6eed80b03e1c58b5fd94e08c83eec176b4c95f5eb560a607234R23), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-349481156959f6eed80b03e1c58b5fd94e08c83eec176b4c95f5eb560a607234R36), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-7a48f7edc8054aad6e6eac7518f6b4bb3b5a33af9687eb69a11ef91dc54ac077R23))
* Added `components/assemblyai` and `components/senta` folders to `pnpm-lock.yaml` file, to indicate installation of new app components as dependencies ([link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR327-R328), [link](https://github.com/PipedreamHQ/pipedream/pull/7321/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4105-R4106))

### CODE REVIEW

File: components/airtable/airtable.app.mjs [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default {
     tableId: {
       type: "string",
       label: "Table",
-      description: "The table ID. If referencing a **Base** dynamically using data from another step (e.g., `{{steps.mydata.$return_value}}`), automatic table options won't work when configuring this step. Please enter a custom expression to specify the **Table**.",
+      description: "The table ID. If referencing a **Base** dynamically using data from another step (e.g., {{steps.trigger.event.metadata.baseId}}), you will not be able to select from the list of Tables, and automatic table options will not work when configuring this step. Please enter a custom expression to specify the **Table**.",
```
Comment (dannyroosevelt) [inline components/airtable/airtable.app.mjs:25]: nit, but is there a reason you removed the code formatting?

File: components/airtable/airtable.app.mjs [hunk 2]
```diff
@@ -40,7 +40,7 @@ export default {
     viewId: {
       type: "string",
       label: "View",
-      description: "The view ID. If referencing a **Table** dynamically using data from another step (e.g., `{{steps.mydata.$return_value}}`), automatic view options won't work when configuring this step. Please enter a custom expression to specify the **View**.",
+      description: "The view ID. If referencing a **Table** dynamically using data from another step (e.g., {{steps.trigger.event.metadata.tableId}}), you will not be able to select from the list of Views for this step. Please enter a custom expression to specify the **View**.",
```
Comment (dannyroosevelt) [inline components/airtable/airtable.app.mjs:43]: nit, but is there a reason you removed the code formatting?

File: components/airtable/airtable.app.mjs [hunk 3]
```diff
@@ -132,6 +132,7 @@ export default {
       type: "boolean",
       label: "Typecast",
       description: "The Airtable API will perform best-effort automatic data conversion from string values if the typecast parameter is `True`. Automatic conversion is disabled by default to ensure data integrity, but it may be helpful for integrating with 3rd party data sources.",
```
Comment (dannyroosevelt) [inline components/airtable/airtable.app.mjs:134]: did you figure out what this actually does / if we actually need it? if it's confusing to you and me, it's likely confusing for users. can we clean up the language in any way?

File: components/airtable/airtable.app.mjs (hunk 1 above)
Comment (malexanderlim) [reply components/airtable/airtable.app.mjs:25]: Ah, sorry I missed that - my paste missed the `. Fixing.

File: components/airtable/airtable.app.mjs (hunk 2 above)
Comment (malexanderlim) [reply components/airtable/airtable.app.mjs:43]: Same as above

File: components/airtable/airtable.app.mjs (hunk 3 above)
Comment (malexanderlim) [reply components/airtable/airtable.app.mjs:134]: It makes a little bit more sense after thinking about it, and revisiting what typecasting does - or type conversion is the process by which you are changing something from one data type to another.

For example, say the Field is expecting a double '10.0', and we pass an integer '10', typecasting would take that integer and convert it into what the Field type is expecting, e.g. 10 --> 10.0. This is also copied directly from their Update record / update API docs, so I think it is okay as is.
https://airtable.com/developers/web/api/update-record

**Review by dannyroosevelt — APPROVED** [review]

**Review by GTFalcao — APPROVED** [review]

**Pull request merged**